bind.go

```diff
@@ -67,7 +67,14 @@ func (b *DefaultBinder) BindQueryParams(c Context, i interface{}) error {
 // See MIMEMultipartForm: https://golang.org/pkg/net/http/#Request.ParseMultipartForm
 func (b *DefaultBinder) BindBody(c Context, i interface{}) (err error) {
 	req := c.Request()
-	if req.ContentLength <= 0 {
```

Comment:
```go
	if req.ContentLength == 0 {
		return
	} else if req.ContentLength < 0 {
		var isChunked bool
		for _, enc := range req.TransferEncoding {
			if enc == "chunked" {
				isChunked = true
				break
			}
		}
		if !isChunked {
			return
		}
	}
```

would be better. we only should check chunked when we are dealing with -1

but I think it would be even better to go back to `if req.ContentLength == 0 {` check